### PR TITLE
fix: all test suites

### PR DESCRIPTION
**Pull request by SRWieZ**
Back ports of my finding in #147

Funny thing, I started that other PR to just fix the test suite. Then my curiosity led me on an adventure in the JavaScript ecosystem and its quirks. After that, I was so deep into it that I couldn't stop.

**Summary of changes:**
- [JS] Created a base mock for `electron` in jest
- [JS] Updated tests that are mocking `electron` to extend the base one
- [JS] Fixed jest.config.ts regex
- [JS] Fixes somes unused imports ans dependencies
- [PHP] tests against L11 and PHP 8.3
- [PHP] Fixed test by requiring `php-bin^5.0.1`  instead of `php-bin^5.0.0` with a weird zip.
- [PHP] Fixed `phpstan` test suite along side all the errors it did find
- [PHP] Added `composer qa`

### CODE REVIEW

**Comment by SRWieZ** [comment]
ALL green! 🎉 ✅

Ready to be reviewed.

File: src/Updater/UpdaterManager.php [hunk 1]
```diff
@@ -52,7 +52,7 @@ public function provider($name = null)
      */
     public function driver($driver = null)
     {
-        return $this->store($driver);
+        return $this->resolve($driver);
```
Comment (simonhamp) [inline src/Updater/UpdaterManager.php:55]: Can you explain this change?

**Review by simonhamp — COMMENTED** [review]
This looks great! I just have one question:

File: src/Updater/UpdaterManager.php (hunk 1 above)
Comment (SRWieZ) [reply src/Updater/UpdaterManager.php:55]: Easy. `store` never existed 

I did searched through git history 😄

**Pull request merged**